src/component/mod.rs

```diff
@@ -1,22 +1,37 @@
 use std::collections::HashMap;
```

Comment:
Why'd you move this to a new file? What is the significance of "mod"?

Reply:
It's just an organizational thing